File: src/promouters/schemas/notifications.py
from datetime import datetime
from uuid import UUID

from pydantic import BaseModel


class NotificationRead(BaseModel):
    id: UUID
    user_id: UUID
    title: str
    body: str
    channel: str
    status: str
    payload: dict | None
    scheduled_at: datetime | None
    sent_at: datetime | None
    read_at: datetime | None
    created_at: datetime
    updated_at: datetime


class NotificationMarkReadResponse(BaseModel):
    success: bool
    notification: NotificationRead


class PhotoReminderDispatchResponse(BaseModel):
    created_count: int
    notification_ids: list[UUID]

File: frontend/src/shared/api/notifications.ts
import { apiRequest } from './http';

export type NotificationStatus = 'pending' | 'sent' | 'read' | 'failed';

export type NotificationRecord = {
  id: string;
  user_id: string;
  title: string;
  body: string;
  channel: string;
  status: NotificationStatus;
  payload: Record<string, unknown> | null;
  scheduled_at: string | null;
  sent_at: string | null;
  read_at: string | null;
  created_at: string;
  updated_at: string;
};

type NotificationReadResponse = {
  success: boolean;
  notification: NotificationRecord;
};

export async function fetchNotifications(accessToken: string) {
  return apiRequest<NotificationRecord[]>('/notifications', {
    method: 'GET',
    accessToken,
  });
}

export async function markNotificationRead(accessToken: string, notificationId: string) {
  const response = await apiRequest<NotificationReadResponse>(`/notifications/${notificationId}/read`, {
    method: 'POST',
    accessToken,
  });

  return response.notification;
}
